File: src/test/testdata/inspections/unsafeIterators/ObjectSet.java
package main;

import com.badlogic.gdx.utils.ObjectSet;

class ObjectSetTest {

  ObjectSet<ObjectSet<String>> set = new ObjectSet<ObjectSet<String>>();

  void test() {
    for (ObjectSet<String> e: <warning>set</warning>) {
      for (String s: <warning>e</warning>) {

      }
    }

    <warning>set.iterator()</warning>;
    <warning><warning>set.iterator()</warning>.next().iterator()</warning>;
  }


}

File: src/test/testdata/inspections/unsafeIterators/ObjectSet.kt
package main

import com.badlogic.gdx.utils.ObjectSet

@Suppress("UNUSED_PARAMETER")
class ObjectSetTestKotlin {

  var set = ObjectSet<ObjectSet<String>>()

  fun test() {
    for (e in <warning>set</warning>) {
      for (s in <warning>e</warning>) {

    }
    }

    <warning>set.iterator()</warning>
    <warning><warning>set.iterator()</warning>.next().iterator()</warning>
  }


}

